### ISSUE TITLE
Make "extended scouting" random number not that random

### ISSUE THREAD

**Issue by vincent-grosbois** [ihhub/fheroes2#2570]
Currently when the option "extended scouting" is on and when the hero has either basic or advance scouting, it will display approximate numbers for monsters and army
However this is randomly generated everytime you right-click, so you get a new number each time which makes it easy to guess what is the correct number.
Also the formula used is made such that even the random interval allow you to deduce the initial amount.
What I propose:
- seed the RNG for scouting based on tile position for monster
- seed the RNG for scouting based on monster id for hero's armies
- change the formula used to compute the random interval such that it is harder to figure out the initial troup size

**Comment by LeHerosInconnu** [comment]
Hello @vincent-grosbois,

> Currently when the option "extended scouting" is on and when the hero has either basic or advance scouting, it will display approximate numbers for monsters and army
> However this is randomly generated everytime you right-click, so you get a new number each time which makes it easy to guess what is the correct number.
> Also the formula used is made such that even the random interval allow you to deduce the initial amount.
> What I propose:
> 
> * seed the RNG for scouting based on tile position for monster
> * seed the RNG for scouting based on monster id for hero's armies
> * change the formula used to compute the random interval such that it is harder to figure out the initial troup size

Your suggestion is welcome, but I find that in its current state this option is a little "light" in terms of the bonus provided.

I have made a slightly more substantial proposal that would give a little more "punch" to the Scouting skill: https://github.com/ihhub/fheroes2/issues/1506. :)

What do you think about it?

**Comment by vincent-grosbois** [comment]
Hello
your discussion is more about new gameplay ideas etc, whereas  I was talking about changing/fixing existing gameplay behavior, they are not incompatible :)

**Comment by LeHerosInconnu** [comment]
Hello @vincent-grosbois,

> Hello
> your discussion is more about new gameplay ideas etc, whereas I was talking about changing/fixing existing gameplay behavior, they are not incompatible :)

Yes, as the current functioning of the option is not really interesting in terms of bonuses (it just gives a slightly more precise approximation than the term used without the skill, nothing really fabulous), I think that it is not necessary to keep this current functioning of the option and that it would be better to completely replace the functioning of the option with the bonuses indicated in my proposal.
Basically, there's no point in wasting time correcting what will be removed later.

For example, concerning the bonuses given for the number of creatures of the wandering troops:

Basic level:
Wandering creatures level 1 to 2: number of creatures, if they will join.

Advanced level:
Wandering creatures level 1 to 4: number of creatures, if they will join.

Expert level:
Wandering creatures level 1 to 6: number of creatures, if they will join.

**Comment by Branikolog** [comment]
Let's leave a completely rework of this skill for a "beyond 1.0" milestone.
For now, it should be switched off on default. 
If @vincent-grosbois wants to try his hand fixing logic of this enhancement, I can only ask not to waste lots of time on this and make some simple solution, as we have more prirority issues. 
After being fixed, it can be useful for newcomers. But I won't complain, if we decide to delete this option at all for now. :)

**Comment by vincent-grosbois** [comment]
I agree with you that it's really "beyond 1.0" . But the issue I have with this is that the option can still be enabled.
Maybe it's best for now to not fix the bugs and to make this option not available in the UI?

**Comment by Branikolog** [comment]
@vincent-grosbois, I'm afraid, that Master developer won't approve such decision. ) Not available option in UI - means deleted option for him. ) 
Let's wait for answers from other members and master dev too. For me, personally, I vote for deleting this code (As I'm pretty sure, it's not good at all to save it for the future improvements.) and for deleting this option from the CONFIG menu.

**Comment by vincent-grosbois** [comment]
Ok I see
I guess the question also applies for all of the "custom settings" that exist and that are not activated by default.  Usually they are more buggy than the rest of the codebase, but at the same time not many people use this

**Comment by LeHerosInconnu** [comment]
Hello @vincent-grosbois,

> Ok I see
> I guess the question also applies for all of the "custom settings" that exist and that are not activated by default. Usually they are more buggy than the rest of the codebase, but at the same time not many people use this

Have you consulted this thread: https://github.com/ihhub/fheroes2/issues/1272?
There is information about the options.
Also, if you found other bugs for the options, please specify the defect so I can indicate it in the first post of the thread. :)

**Comment by vincent-grosbois** [comment]
Hello
actually this idea is also needed for Thieves Guild, so it's not just for non-OG features
